File: oscexchange/src/main/java/ac/at/fhstp/digitech/oscexchange/OSCParseException.java
package ac.at.fhstp.digitech.oscexchange;

/**
 * An exception that occurs when invalid OSCArgs are force-parsed
 */
public final class OSCParseException extends Exception {

    /**
     * The args that were attempted to parse
     */
    public final OSCArgs args;

    public OSCParseException(OSCArgs args) {
        super("Could not parse OSC args!");
        this.args = args;
    }
}

File: oscexchange/src/main/java/ac/at/fhstp/digitech/oscexchange/OSCParseException.kt
package ac.at.fhstp.digitech.oscexchange

/**
 * An exception that occurs when invalid OSCArgs are force-parsed
 */
class OSCParseException(
    /**
     * The args that were attempted to parse
     */
    @PublicApi val args: OSCArgs
) : Exception("Could not parse OSC args!")